design-documents/frontend/technical-vision.md

```diff
@@ -0,0 +1,140 @@
+
+> We test browser support with a test suite for each feature. A browser is considered compliant only if it passes all the tests, so this is a reasonably comprehensive test rather than a naïve feature-detect.
+
+_TODO: Investigate building two separate [TypeScript][] builds, one that compiles down to ES5 and another lighter-weight modern build, specifically for modern browsers. Depending on how well Polyfill.io works, the modern build might be all we need._
+
+## React UI Component Library
+
+_Consistency, Productivity, Scalability, Maintainability, Accessibility._
+
+We should lean against downloading various UI components from different sources for the following reasons:
+
+- **Inconsistent** implementations, documentation, theming and customization.
+- Limited **accessibility** and support.
+
+[KendoReact | UI for React Developers](https://www.telerik.com/kendo-react-ui/) is a great option for those wishing to get React UI components from the same source. Still, there will be times when you need to download a 3rd-party package.
+
+### Downloading a 3rd-party package
+
+You may have to download a 3rd party component. When this happens, consider the following criteria before making a final choice:
+
+1. First thing's first – what is the license?
+   - Is it open source?
+   - You may need to reach out to your legal department if things are a bit blurry.
+1. Does the package have a bright future?
+   - Popularity – are there a significant number of weekly downloads?
+   - Is the downloads line graph going up or down?
+1. Look for tests, ideally with 100% test coverage.
+1. Look at install size with [Package Phobia](https://packagephobia.now.sh/).
+1. Does it have good/clear documentation?
+1. Is it currently maintained?
+   - Look at the number of issues vs. pull requests.
+   - How long have these issues and PRs been open?
+   - When was the last comment made on them and by whom?
+   - **Reliability** – how serious are the issues?
+1. Search and compare this package with other similar packages to gauge which package is the best choice and feel free to reach out to other team members for guidance.
+
+### Resources
+- [spectrum-css](http://opensource.adobe.com/spectrum-css/2.6.0/docs/)
+
+## Styles
+
+Bundle optimization works best when CSS Modules are imported directly into the JavaScript components that use them (a better [TypeScript][] experience is being talked about in [this CRA issue](https://github.com/facebook/create-react-app/issues/5677)). The coupling here is a bit unfortunate for some projects (e.g., libraries), but pretty standard for applications. If there is a specific reason for ultimate **portability** in your project, consider the [container paradigm][] to decouple the styles from the components, each component defining its own styles via a styles interface.

```

Comment:
Think it would make sense to remove the reference to the container paradigm here. From what I'm seeing, it's going to quickly disappear in lieu of similar things that can be done with hooks.

Opinion: For now, this section should stick to the _what_, rather than the _how_.

Reply:
I like that suggestion.

Reply:
Addressed in 4f625a47b50e0c0ac67ccbf5fed03eff17b94d40